packages/astro/src/content/vite-plugin-content-virtual-mod.ts

```diff
@@ -20,98 +21,177 @@ import {
 	getEntryType,
 	getExtGlob,
 	type ContentLookupMap,
-	type ContentPaths,
 } from './utils.js';
+import type { AstroPluginMetadata } from '../vite-plugin-astro/index.js';
+import { isServerLikeOutput } from '../prerender/utils.js';
 
 interface AstroContentVirtualModPluginParams {
 	settings: AstroSettings;
+	fs: typeof nodeFs
 }
 
 export function astroContentVirtualModPlugin({
 	settings,
+	fs,
 }: AstroContentVirtualModPluginParams): Plugin {
+	let IS_DEV = false;
+	const IS_SERVER = isServerLikeOutput(settings.config);
+	return {
+		name: 'astro-content-virtual-mod-plugin',
+		enforce: 'pre',
+		configResolved(config) {
+			IS_DEV = config.mode === 'development'
+		},
```

Comment:
`settings` already contains the `config`; why do we need `configResolved` in this case?